### PR TITLE
Add pad transform, string to int transform

### PR DESCRIPTION
**Pull request by ebsmothers**
Test plan:

Add unit tests

```
python3 -m pytest -v -k 'pad_transform or str_to_int'
====================================================================================================== test session starts =======================================================================================================
platform linux -- Python 3.9.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /data/home/ebs/miniconda/envs/torchtext/bin/python3
cachedir: .pytest_cache
rootdir: /data/home/ebs/torchtext, configfile: pytest.ini, testpaths: test/
plugins: pythonpath-0.7.4, cov-3.0.0
collected 1091 items / 1087 deselected / 4 selected

test/test_transforms.py::TestTransforms::test_pad_transform PASSED                                                                                                                                                         [ 25%]
test/test_transforms.py::TestTransforms::test_pad_transform_jit PASSED                                                                                                                                                     [ 50%]
test/test_transforms.py::TestTransforms::test_str_to_int_transform PASSED                                                                                                                                                  [ 75%]
test/test_transforms.py::TestTransforms::test_str_to_int_transform_jit PASSED                                                                                                                                              [100%]

============================================================================================== 4 passed, 1087 deselected in 10.57s ===============================================================================================
```

### CODE REVIEW

File: torchtext/transforms.py [hunk 1]
```diff
@@ -221,6 +223,33 @@ def forward(self, input: Any) -> Any:
...
+        super().__init__()
+        self.max_length = max_length
+        self.pad_value = pad_value
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        max_encoded_length = x.size(-1)
+        if max_encoded_length < self.max_length:
+            pad_amount = self.max_length - max_encoded_length
+            x = torch.nn.functional.pad(x, (0, pad_amount), value=self.pad_value)
+        return x
```
Comment (parmeet) [inline torchtext/transforms.py:237]: could you please add doc-strings for `__init__` and `forward` methods? Also update the the [rst file](https://github.com/pytorch/text/blob/main/docs/source/transforms.rst) to update the documentation.

File: torchtext/transforms.py [hunk 2]
```diff
@@ -221,6 +223,33 @@ def forward(self, input: Any) -> Any:
...
+
+
+class StrToIntTransform(Module):
```
Comment (parmeet) [inline torchtext/transforms.py:240]: Same comment for doc-strings :)

File: docs/source/transforms.rst [hunk 3]
```diff
@@ -71,3 +71,17 @@ Sequential
 .. autoclass:: Sequential
 
    .. automethod:: forward
+
+PadTransform
+----------
```
Comment (mthrok) [inline docs/source/transforms.rst:76]:
```suggestion
------------
```

File: docs/source/transforms.rst [hunk 4]
```diff
@@ -71,3 +71,17 @@ Sequential
...
+
+.. autoclass:: PadTransform
+
+   .. automethod:: forward
+
+StrToIntTransform
+----------
```
Comment (mthrok) [inline docs/source/transforms.rst:83]:
```suggestion
-----------------
```

File: test/test_transforms.py [hunk 5]
```diff
@@ -205,6 +205,76 @@ def test_add_token(self):
     def test_add_token_jit(self):
         self._add_token(test_scripting=True)
 
+    def _pad_transform(self, test_scripting):
```
Comment (mthrok) [inline test/test_transforms.py:208]: Can you add a brief description of what this test does?
In the form of "(under some condition), an API produces a result that satisfies this property"

**Review by parmeet — APPROVED** [review]
LGTM! Thanks for upstreaming the transforms from torchMM. I think once we land this, the next step could be use the transforms directly from torchtext and remove them from torchMM :)

**Pull request merged**